public/app/features/alerting/unified/utils/k8s/utils.ts

```diff
@@ -30,3 +31,33 @@ type Entity = {
  */
 export const isK8sEntityProvisioned = (item: Entity) =>
   item.metadata.annotations?.[PROVENANCE_ANNOTATION] !== PROVENANCE_NONE;
+
+type EntityToCheck = {
+  metadata?: IoK8SApimachineryPkgApisMetaV1ObjectMeta;
+};
+
+/**
+ * Different permissions that we expect to see in the k8s metadata annotations
+ * under the `grafana.com/access/` key
+ * */
+type K8sPermission = 'canWrite' | 'canAdmin' | 'canDelete';
+
+export const ANNOTATION_PREFIX_ACCESS = 'grafana.com/access/';
+
+/**
+ * Checks annotations on a k8s entity to see if the requesting user has the required permission
+ */
+const getContactPointPermission = (k8sEntity: EntityToCheck, permission: K8sPermission) =>
+  k8sEntity.metadata?.annotations?.[ANNOTATION_PREFIX_ACCESS + permission] === 'true';
+
+export const canEditEntity = (k8sEntity: EntityToCheck) => getContactPointPermission(k8sEntity, 'canWrite');
+
+export const canAdminEntity = (k8sEntity: EntityToCheck) => getContactPointPermission(k8sEntity, 'canAdmin');
+
+export const canDeleteEntity = (k8sEntity: EntityToCheck) => getContactPointPermission(k8sEntity, 'canDelete');
+
+/** Annotation key that indicates how many notification policy routes are using this entity */
+export const ANNOTATION_INUSE_ROUTES = 'grafana.com/inUse/routes';
+
+/** Annotation key that indicates how many alert rules are using this entity */
+export const ANNOTATION_INUSE_RULES = 'grafana.com/inUse/rules';
```

Comment:
What do you think about grouping known annotation keys in a one enum object?
Like
```ts
enum K8sAnnotations {
	InUseRoutes = 'grafana.com/inUse/routes',
	InUseRules = 'grafana.com/inUse/rules'
}

I feel it will be clear to group them together than keeping them as separate consts. Probably `provenance` could also fit there

Reply:
I like it :+1: Will make this change

Reply:
I made changes - but didn't entirely move `provenance` over as I didn't want to change the other uses and grow this PR any further. I marked the old annotation as deprecated so the few extra places will be more easily spotted and cleaned up after